CMakeLists.txt

```diff
@@ -16,10 +16,24 @@ find_package(OpenMP)
 add_subdirectory(src/PPN-microbench)
 add_subdirectory(main)
 
+# Setup architecture for assembly
+
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(arm|aarch64)")
+    message(STATUS "Configuring for ARM")
+    target_compile_options(PPN-microbench PRIVATE -march=armv8-a -mtune=cortex-a72)

```

Comment:
Why not use `native` for `-march`, as you do for x86? Tuning for a specific CPU by default while you are developing a portable tool seems to be conter-intuitive too. 

Also, on Arm you should prefer `-mcpu=native` to `-march` (and `-mtune` won't be needed anymore). See [this](https://community.arm.com/arm-community-blogs/b/tools-software-ides-blog/posts/compiler-flags-across-architectures-march-mtune-and-mcpu).